index.js

```diff
@@ -198,7 +208,15 @@ export default {
     // Args:
     //     web3in: web3 instance
     //     jrpcUrl: QuarkChain JSON RPC endpoint (e.g., http://localhost:38391)
-    const web3http = new Web3(new Web3.providers.HttpProvider(jrpcUrl));
+    const web3http = new Web3(new Web3HttpProvider(jrpcUrl));
+
+    let tmpWeb3in;
```

Comment:
seems only `web3in.eth` is affected other than `metaMaskSignTyped`. suggest:

```javascript
let web3eth;
if (web3in.version && !web3in.version.api) {
    // web3in version 1.x
    web3eth = (new Web3(web3in.currentProvider)).eth;
} else {
    web3eth = web3in.eth;
}
```
for `metaMaskSignTyped`, you can get rid off its argument `tmpweb3in` by directly passing `from` address along with  `web3in`

Reply:
Got it!